doc/rtd/explanation/about-cloud-config.rst

```diff
@@ -0,0 +1,167 @@
+.. _about-cloud-config:
+
+About the cloud-config file
+***************************
+
+The ``#cloud-config`` file is a type of user data that cloud-init can consume
+to automatically set up various aspects of the system. It is the preferred way
+to pass this type of configuration to cloud-init, and is commonly referred to
+as "cloud config".
+
+Note that networks are not configurable via the ``#cloud-config`` file because
+:ref:`network configuration <network_config>` comes from the cloud.
+
+How do I create a cloud-config file?
+====================================
+
+The cloud-config file uses `YAML version 1.1`_. The file is composed of a
+**header** and one or more **modules**.
+
+* **The header**:
+  The first line **must start** with ``#cloud-config``. This line identifies
+  the file to cloud-init and ensures that it will be processed as intended.
+
+* **The modules**:
+  After the header, every aspect of the system's configuration is controlled
+  through specific cloud-init modules.
+
+Each module is declared through the use of its **top-level key**, and the
+configuration options are set using key-value pairs (or list keys) according
+to the config schema. It follows this general format:
+
+.. code-block:: yaml
+
+   #cloud-config
+   top-level-key:
+     config-key-1: config-value-1
+     config-key-2: config-value-2
+     list-key:
+     - list-value-1
+       additional-list-value-1
+     - list-value-2
+
+Let us consider a real example using the :ref:`Keyboard <mod_cc_keyboard>`
+module. The top-level key for this module is ``keyboard:``, and beneath that
+are the various configuration options for the module shown as key-value pairs:
+
+.. code-block:: yaml
+
+   #cloud-config
+   keyboard:
+     layout: us
+     model: pc105
+     variant: nodeadkeys
+     options: compose:rwin
+
+A full list of modules can be found :ref:`on our modules page <modules>`. This
+list also shows the valid schema keys for every module, and YAML examples.
+
+Module ordering
+---------------
+
+The order of the ``cloud-config`` keys is unimportant and modules can be
+written in any order.
+
+.. note::
+   The :ref:`Users and Groups <mod_cc_users_groups>` module is a special case
```

Comment:
Do we need this note?

There are other examples where ordering matters too - any place in the schema where a list is used rather than a k/v pair is likely to have an ordering effect.